rasa/core/policies/ted_policy.py

```diff
@@ -602,11 +602,11 @@ def _create_optional_event_for_entities(
         if tracker.latest_action_name != ACTION_LISTEN_NAME:
             # entities belong to the last user message
             # a user message is always followed by action listen
-            return None
+            return
```

Comment:
@tabergma in this method `prediction_output` will contain the batch of 2 inputs. I'd say we should add entities only if e2e prediction is chosen. Do we do that? Sorry, I missed it in your PR

Reply:
Inside TED I only check if `self.config[ENTITY_RECOGNITION]` is set to `True`. So no, we don't have that right now.

Reply:
hm, that needs to be fixed then, how does it extract entities from the batch of size 2?

Reply:
Not sure what you mean. If the model was trained with entity recognition, it will automatically predict entities on incoming user utterances. We extract any entities if they are present.

Reply:
yes, but at prediction time due to e2e, the output will contain affectively 2 user utterance. And we need to predict entities only if e2e won

Reply:
I updated it in this PR, could you please take a look?